### DISCUSSION TITLE
Facilitator / Non-player access to game state

### DISCUSSION THREAD

**Discussion by sridharraman** [boardgameio/boardgame.io#1151] [Q&A]
Hi

We are building a multi-player game where players make moves simultaneously. That's been fine so far. I was wondering what would be the best practice to create a "dashboard" for facilitator/non-player to view the game state. Is that possible?

Going through the documentation, I came up with a couple of approaches:
1. Using [Storage](https://boardgame.io/documentation/#/storage) to store the game state in a database (flatfile / postgres) and then building another small web app that accesses the db for the Dashboard
2. Some way of creating an "observer" as a non-player in the game and give them read-only access to the game state

Is there a recommended way of doing this? Have I missed any other approach?

**Comment by psalaets** [comment]
What did you end up going with?

I haven't tried any of this yet but just throwing out some ideas.

- Add the facilitator as a player and use a custom turn order so the facilitator never gets a turn. Maybe they are put into a special stage that contains facilitator-only moves.
- Inside `playerView` detect the facilitator and return the data needed to populate the dashboard.
- I believe a user can connect to a game without providing `playerID` nor `credentials` and they'll essentially be a readonly observer.

**Reply by sridharraman** [reply]
I ended up going with a variant of the first idea. It's a 5-player game, so I created an additional player (id=0) and made this as the Facilitator. Additionally, our game is multi-player with a timer. So I used the Facilitator as the only player who would have access to the timer.

An issue that we are facing (not entirely related to this) is the stale id error thrown by bgio. Any idea how this can be fixed?

**Reply by psalaets** [reply]
> An issue that we are facing (not entirely related to this) is the stale id error thrown by bgio. Any idea how this can be fixed?

No sorry. I haven't encountered that error.
